File: ios/GaelGrounds/Services/AvatarService.swift
import Foundation
import UIKit
import Supabase

enum AvatarService {
    private static let bucket = "avatars"

    enum UploadError: Error {
        case invalidImage
    }

    static func upload(rawImageData: Data, userId: UUID) async throws -> String {
        guard let uiImage = UIImage(data: rawImageData),
              let jpegData = uiImage.jpegData(compressionQuality: 0.8) else {
            throw UploadError.invalidImage
        }

        let path = "\(userId.uuidString)/\(UUID().uuidString).jpg"
        try await Supa.client.storage.from(bucket).upload(
            path,
            data: jpegData,
            options: FileOptions(contentType: "image/jpeg")
        )
        return try Supa.client.storage.from(bucket).getPublicURL(path: path).absoluteString
    }
}

File: android/app/src/main/kotlin/ie/gaelgrounds/app/data/service/AvatarService.kt
package ie.gaelgrounds.app.data.service

import ie.gaelgrounds.app.data.Supa
import io.github.jan.supabase.storage.storage
import java.util.UUID

/** Mirrors ios/GaelGrounds/Services/AvatarService.swift. */
object AvatarService {
    private const val BUCKET = "avatars"

    suspend fun upload(jpegBytes: ByteArray, userId: String): String {
        val path = "$userId/${UUID.randomUUID()}.jpg"
        Supa.client.storage.from(BUCKET).upload(path, jpegBytes)
        return Supa.client.storage.from(BUCKET).publicUrl(path)
    }
}
